compiler/optimizer/VPHandlers.cpp

```diff
@@ -1376,7 +1376,15 @@ TR::Node *constrainAnyIntLoad(OMR::ValuePropagation *vp, TR::Node *node)
                TR::VPConstString *constString = baseVPConstraint->getClassType()->asConstString();
 
                uintptr_t offset = vp->comp()->target().is64Bit() ? (uintptr_t)index->getUnsignedLongInt() : (uintptr_t)index->getUnsignedInt();
-               uintptr_t chIdx = (offset - (uintptr_t)TR::Compiler->om.contiguousArrayHeaderSizeInBytes()) / 2;
+               uintptr_t chIdx = 0;
+               if (array->isDataAddrPointer())
```

Comment:
It may be better to explain the reasoning here via some comments (including the original "divide by 2" logic).

Reply:
Added comment "Get index from offset of a 2-byte element array"

Reply:
Thanks, but I meant also why the below math is different for data address pointer and why it works out in the "if" and "else" cases.